plugins/inputs/mysql/v2/convert.go

```diff
@@ -0,0 +1,94 @@
+package v2
+
+import (
+	"bytes"
+	"database/sql"
+	"fmt"
+	"strconv"
+)
+
+type ConversionFunc func(value sql.RawBytes) (interface{}, error)
+
+func ParseInt(value sql.RawBytes) (interface{}, error) {
+	v, err := strconv.ParseInt(string(value), 10, 64)
+
+	// Ignore ErrRange.  When this error is set the returned value is "the
+	// maximum magnitude integer of the appropriate bitSize and sign."
+	if err, ok := err.(*strconv.NumError); ok && err.Err == strconv.ErrRange {
+		return v, nil
+	}
+
+	return v, err
+}
+
+func ParseBoolAsInteger(value sql.RawBytes) (interface{}, error) {
+	if bytes.EqualFold(value, []byte("YES")) || bytes.EqualFold(value, []byte("ON")) {
+		return int64(1), nil
+	}
+
+	return int64(0), nil
+}
+
+func ParseGTIDMode(value sql.RawBytes) (interface{}, error) {
+	// https://dev.mysql.com/doc/refman/8.0/en/replication-mode-change-online-concepts.html
+	v := string(value)
+	switch v {
+	case "OFF":
+		return int64(0), nil
+	case "ON":
+		return int64(1), nil
+	case "OFF_PERMISSIVE":
+		return int64(0), nil
+	case "ON_PERMISSIVE":
+		return int64(1), nil
+	default:
+		return nil, fmt.Errorf("unrecognized gtid_mode: %q", v)
+	}
+}
+
+var GlobalStatusConversions = map[string]ConversionFunc{
+	"ssl_ctx_verify_depth": ParseInt,
+}
+
+var GlobalVariableConversions = map[string]ConversionFunc{
+	"gtid_mode": ParseGTIDMode,
+}
+
```

Comment:
gtid_mode, ssl_ctx_verify_depth, ssl_verify_depth are the fields that have caused issues for us but I wonder if there are others that will be a problem in the future.  Were you able to find a list of these changes in mysql API docs for instance?  I wonder if adding them one at a time will be a whack-a-mole style headache that we could prevent by finding all the changes before they turn into issues.

Reply:
I agree this is going to be whack-a-mole, but I just checked and I don't see a document describing changes in this table.  I believe the table is designed more for interactive use than how we are using it.

Perhaps it would make sense to whitelist in the known variables instead of trying to parse them dynamically?  I also have some concerns about breaking compatibility if I get too aggressive changing the parse behavior of the fields though.